gramps/gen/utils/alive.py

```diff
@@ -89,234 +96,420 @@ def __init__(
+                # create a range set
+                birth_range = list(Date.EMPTY + Date.EMPTY)
+                birth_range[Date._POS_YR] = min_birth_year
+                birth_range[Date._POS_RYR] = max_birth_year
+                birth_date.set(modifier=Date.MOD_RANGE, value=tuple(birth_range))
+            else:
+                if min_birth_year:
+                    birth_date.set_yr_mon_day(min_birth_year, 1, 1)
+                    birth_date.set_modifier(Date.MOD_AFTER)
+                elif max_birth_year:
+                    birth_date.set_yr_mon_day(max_birth_year, 12, 31)
+                    birth_date.set_modifier(Date.MOD_BEFORE)
+            birth_date.recalc_sort_value()
+
+        # If we have no death date but we know death has happened then
+        # we set death range somewhere between birth and yesterday.
+        # otherwise we assume MAX years after birth
+        if death_date is None:
+            if birth_date and birth_date.is_valid():
+                death_date = Date(birth_date)
+                max_death_date = birth_date.copy_offset_ymd(
+                    year=self.MAX_AGE_PROB_ALIVE
+                )
+                if known_to_be_dead:
+                    if max_death_date.match(Today(), ">="):
+                        max_death_date = Today()
+                        max_death_date.set_yr_mon_day_offset(
+                            day=-1
+                        )  # make it yesterday
+                    # range start value stays at birth date
+                    death_date.set_modifier(Date.MOD_RANGE)
+                    death_date.set_text_value("")
+                    death_date.set2_yr_mon_day(
+                        max_death_date.get_year(),
+                        max_death_date.get_month(),
+                        max_death_date.get_day(),
+                    )
+                    explain_death = _("birth date and known to be dead")
+                else:
+                    death_date.set_yr_mon_day_offset(year=self.MAX_AGE_PROB_ALIVE)
+                    explain_death = _("birth date")
+                death_date.recalc_sort_value()
+            else:
+                death_date = Date()
+
+        # at this stage we should have valid dates for both birth and death,
+        #  or else both are zero (if None then it's a bug).
+        if explain_birth_max == "":
+            explanation = _("birth: ") + explain_birth_min
+        else:
+            explanation = (
+                _("birth: ") + explain_birth_min + _(" and ") + explain_birth_max
+            )
+        explanation += _(", death: ") + explain_death
+        explanation = "2ND + " + explanation
+        if birth_date.is_valid() and death_date.is_valid():
+            return (birth_date, death_date, explanation, person)
+
+        # have finished immediate family, so try spouse, as the
+        # remaining person (probably) of this generation ..
+
+        def spouse_test(passnum=1):
```

Comment:
Given that `passnum` only appears to be used to determine the value of `immediate_fam_only`, you could have have `immediate_fam_only` as the method parameter. Would this be clearer than `passnum`?

Reply:
Yes, you are right, although I would still use a different variable name because `immediate_fam_only` is a parameter in the outer function. Using the same name in overlapping scopes leaves me a bit uneasy.